lib/TransferConfigurations.cs

```diff
@@ -184,7 +184,8 @@ internal long MaximumCacheSize
         /// <param name="newBlockSize"></param>
         internal void UpdateMaximumCacheSize(int newBlockSize)
         {
-            this.MaximumCacheSize = (long)3 * newBlockSize * this.ParallelOperations;
+            // TODO: This is no longer a very good way to set the max cache size
+            this.MaximumCacheSize = (long)10 * newBlockSize * this.ParallelOperations;
```

Comment:
This is the most magic thing need further efforts! Any magic number could cause regression, wish further cooperation on making this thing generic!

Reply:
This change, or a better generalization, is needed to capture the new perf improvements. What kind of scenarios do we need to test to ensure we catch any possible regressions? Is there anything you'd like me to run before the end of the week?